modules/core/src/main/scala/com.snowplowanalytics.iglu/client/validator/CirceValidator.scala

```diff
@@ -119,16 +129,67 @@ object CirceValidator extends Validator[Json] {
       }
     } yield ()
 
-  private def fromValidationMessage(m: ValidationMessage): ValidatorReport =
-    ValidatorReport(m.getMessage, m.getPath.some, m.getArguments.toList, m.getType.some)
+  /**
+   * Extract custom formatted message from ValidationMessage.
+   * This is used for both data validation errors and schema validation errors.
+   */
+  private def getCustomMessage(m: ValidationMessage): String = {
```

Comment:
I wonder if this whole function could be replaced with:

```scala
private def getCustomMessage(m: ValidationMessage): String =
  fromValidationMessage(m).message
```

I know that's a bit wasteful in a way, because it calculates all fields of the `ValidatorReport` and then ignores most of those fields.  But, I _think_ this method is only called when validating schemas.  And only if a schema is invalid.

When you think about it... it's quite rare this function will ever get executed.  We don't expect our pipelines to encounter many invalid schemas ever!  So I think simplicity is more important than an an optimized implementation for this function.

Please check my thinking before agreeing!

Reply:
comment was outdated, I agree with your point and I'll prefer simplicity 👍